### PR TITLE
Cleanup: Fix grammar in error message, also improve readability.

### PR DESCRIPTION
**Pull request by mayankshriv**
* Fixed grammar.
* Now logging table name first, before the list of segments, for better readability.

### CODE REVIEW

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java [hunk 1]
```diff
@@ -243,7 +243,7 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       Set<String> unavailableSegments = entry.getValue();
       numUnavailableSegments += unavailableSegments.size();
       brokerResponse.addException(QueryException.getException(QueryException.SERVER_SEGMENT_MISSING_ERROR,
-          String.format("Find unavailable segments: %s for table: %s", unavailableSegments, tableName)));
+          String.format("Found unavailable segments for table {%s}: {%s}", tableName, unavailableSegments)));
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java:246]:
```suggestion
          String.format("Found unavailable segments for table %s: %s", tableName, unavailableSegments)));
```

Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java:246]: Consider adding number of unavailable segments, and only log up to 10 segments. If there are more than 10 segments unavailable, append `...` after the list

Comment (mayankshriv) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java]: Done @Jackie-Jiang. Also, given that error path will be very infrequent, I felt it was ok to use stream way to build the string as opposed to StringBuilder and for loop.

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java [hunk 2]
```diff
@@ -243,7 +244,8 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       Set<String> unavailableSegments = entry.getValue();
       numUnavailableSegments += unavailableSegments.size();
       brokerResponse.addException(QueryException.getException(QueryException.SERVER_SEGMENT_MISSING_ERROR,
-          String.format("Find unavailable segments: %s for table: %s", unavailableSegments, tableName)));
+          String.format("Found unavailable segments for table %s: %s", tableName,
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java:247]: Suggest adding the count `Found %d unavailable segments ...`

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

**Pull request merged**